onnxruntime/core/session/inference_session.cc

```diff
@@ -528,124 +528,165 @@ common::Status InferenceSession::Load(const std::basic_string<T>& model_uri) {
   return Status::OK();
 }
 
+#endif  // !defined(ORT_MINIMAL_BUILD)
+
 common::Status InferenceSession::Load(const std::string& model_uri) {
+  std::string model_type = session_options_.GetConfigOrDefault(ORT_SESSION_OPTIONS_CONFIG_LOAD_MODEL_FORMAT, "");
+  bool has_explicit_type = !model_type.empty();
+
+  if ((has_explicit_type && model_type == "ORT") ||
+      (!has_explicit_type && inference_session_utils::IsOrtFormatModel(model_uri))) {
+    return LoadOrtModel(model_uri);
+  }
+
+#if !defined(ORT_MINIMAL_BUILD)
   if (is_model_proto_parsed_) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
                            "ModelProto corresponding to the model to be loaded has already been parsed. "
                            "Invoke Load().");
   }
 
   return Load<char>(model_uri);
+#else
+  return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "ONNX format model is not supported in this build.");
+#endif
 }
 
 #ifdef _WIN32
 common::Status InferenceSession::Load(const std::wstring& model_uri) {
+  std::string model_type = session_options_.GetConfigOrDefault(ORT_SESSION_OPTIONS_CONFIG_LOAD_MODEL_FORMAT, "");
+  bool has_explicit_type = !model_type.empty();
+
+  if ((has_explicit_type && model_type == "ORT") ||
+      (!has_explicit_type && inference_session_utils::IsOrtFormatModel(model_uri))) {
+    return LoadOrtModel(model_uri);
+  }
+
+#if !defined(ORT_MINIMAL_BUILD)
   if (is_model_proto_parsed_) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
                            "ModelProto corresponding to the model to be loaded has already been parsed. "
                            "Invoke Load().");
   }
 
   return Load<PATH_CHAR_TYPE>(model_uri);
+#else
+  return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "ONNX format model is not supported in this build.");
+#endif
 }
 #endif
 
-common::Status InferenceSession::Load(const ModelProto& model_proto) {
+common::Status InferenceSession::Load(const void* model_data, int model_data_len) {
```

Comment:
GitHub diff for this is confusing. Swapped the order of `Load(const ModelProto& model_proto)` and `Load(const void* model_data, int model_data_len)` so the ORT_MINIMAL_BUILD exclusion was grouped better. 

Only real change is to do the ORT format bytes detection in the load from bytes method.